File: Lab2/c/validations.c
#include <ctype.h>

int clean_validate(char *input) {
    int i = 0; // Lector
    int j = 0; // Escritor

    while (input[i] != '\0') {
        if (isalpha((unsigned char)input[i]) || isspace((unsigned char)input[i]) || (unsigned char)input[i] > 127) {
            input[j] = input[i];
            j++;
        }
        i++;
    }
    
    input[j] = '\0'; 
    return (j > 0); 
}

File: Lab2/go/validations.go
package main

func cleanValidate(data []byte) int {
    j := 0
    for i := 0; i < len(data); i++ {
        c := data[i]
        if (c >= 'a' && c <= 'z') || (c >= 'A' && c <= 'Z') || c == ' ' || c > 127 {
            data[j] = data[i]
            j++
        }
    }
    return j
}